### PR TITLE
fix(query): add circe decoder to labelvalues and series-matcher query results

### PR DESCRIPTION
**Pull request by sherali42**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?

- add circe decoder to labelvalues and series-matcher query results to support multi-partition queries
- bug fix to return an array of tag values, instead of map, when the metadata query is only for single column

### CODE REVIEW

**Review by amolnayak311 — APPROVED** [review]
LGTM

File: query/src/main/scala/filodb/query/PromCirceSupport.scala [hunk 1]
```diff
@@ -41,29 +43,39 @@ object PromCirceSupport {
...
+        case None =>
+          if (c.value.isString)
+            for { label <- c.as[String] } yield LabelSampl(label)
+          else
+            for { metadataMap <- c.as[Map[String, String]] } yield MetadataMapSampl(metadataMap)
+      }
+    }
+  }
+
+  implicit val decodeFoo: Decoder[DataSampl] = new Decoder[DataSampl] {
```
Comment (tjackpaul) [inline query/src/main/scala/filodb/query/PromCirceSupport.scala:65]: do we really have to use this variable name as `decodeFoo` ?

File: query/src/test/scala/filodb/query/exec/RemoteMetadataExecSpec.scala [hunk 2]
```diff
@@ -31,27 +39,94 @@ class RemoteMetadataExecSpec extends AnyFunSpec with Matchers with ScalaFutures
...
     .thenRespondWrapped(Future {
-      Response(Right(Right(MetadataSuccessResponse(Seq(LabelSampl(Seq("http_req_total", "http_resp_time"))), "success", Option.empty, Option.empty))), StatusCodes.PartialContent, "", Nil, Nil)
+      Response(Right(Right(MetadataSuccessResponse(Seq(LabelSampl("http_req_total"), LabelSampl("http_resp_time")), "success", Option.empty, Option.empty))), StatusCodes.PartialContent, "", Nil, Nil)
     })
     .whenRequestMatches(request =>
       request.uri.path.startsWith(List("api","v1","series"))
     )
     .thenRespondWrapped(Future {
-      Response(Right(Right(MetadataSuccessResponse(Seq(MetadataSampl(Map(("job" -> "myCoolService"), ("unicode_tag" -> "uniμtag")))), "success", Option.empty, Option.empty))), StatusCodes.PartialContent, "", Nil, Nil)
+      Response(Right(Right(MetadataSuccessResponse(Seq(MetadataMapSampl(Map(("job" -> "myCoolService"), ("unicode_tag" -> "uniμtag")))), "success", Option.empty, Option.empty))), StatusCodes.PartialContent, "", Nil, Nil)
```
Comment (tjackpaul) [inline query/src/test/scala/filodb/query/exec/RemoteMetadataExecSpec.scala:124]: Can you also add a test case with empty array for `LabelSampl` and `MetadataMapSampl`.

**Review by tjackpaul — APPROVED** [review]
thanks for the PR @sherali42 .. left minor comments

File: query/src/main/scala/filodb/query/PromCirceSupport.scala (hunk 1 above)
Comment (sherali42) [reply query/src/main/scala/filodb/query/PromCirceSupport.scala]: renamed it

File: query/src/test/scala/filodb/query/exec/RemoteMetadataExecSpec.scala (hunk 2 above)
Comment (sherali42) [reply query/src/test/scala/filodb/query/exec/RemoteMetadataExecSpec.scala:130]: added testcases for empty results
